src/num/flpmath.c

```diff
@@ -2992,3 +2992,53 @@ void md_zfftmod(unsigned int D, const long dims[D], complex float* optr, const c
 
 	md_zfftmod2(D, dims, strs, optr, strs, iptr, inv, phase);
 }
+
+extern void md_calc_fmac3_dims(long N, long squash, long dims1[N], long dims2[N], long out_dims[N]) 
+{
+	long dims[N];
+	for (int i = 0; i < N; i++) {
+		assert((dims1[i] == dims2[i]) || (1 == dims1[i]) || (1 == dims2[i]));
+		dims[i] = (1 == dims1[i]) ? dims2[i] : dims1[i];
+	}   
+	md_select_dims(N, ~squash, out_dims, dims);
+}
+
+extern void md_fmac3(long N, long dims1[N], complex float* input1, long dims2[N], complex float* input2, long out_dims[N], complex float* out_data) 
```

Comment:
In general the `md_` functions that take complex floats as inputs should start with `md_z`.

Now, I'm not sure I agree with naming this `fmac3`, since its interface is the same as `fmac`, i.e., it doesn't take strides as input. Some thoughts:

- Our convention is to have the output args come before the input args in functions. 
- I think you should rename `md_calc_fmac3_dims` to `md_calc_fmac_dims`
- Your code could call `md_calc_fmac_dims` and `calc_strides`
- Then you ccould call `md_zfmac2` and `md_zfmacc2` directly, without defining a new function `md_fmac3`.

Sorry if I misunderstood the use of this function.